omics/cli/run_analyzer/__main__.py

```diff
@@ -522,11 +552,33 @@ def tocsv(val):
 
             writer = csv.writer(out, lineterminator="\n")
             writer.writerow(formatted_headers)
+            config = {}
+
             for res in resources:
                 add_metrics(res, resources, pricing, headroom)
                 metrics = res.get("metrics", {})
+                if res['type'] == 'run':
+                    omics = session.client("omics")
+                    wfid = res['workflow'].split('/')[-1]
+                    engine = omics.get_workflow(id=wfid)['engine']
+                if res['type'] == 'task':
+                    task_name = res['name'].split(" ")[0]
```

Comment:
Will this always work? Do we know that our engines can't produce a task name like "[my workflow task]" with spaces?

It might make this more future proof to have a constant for the task name split string because if we need to change it due to new engines, this line of code doesn't advertise it's intent. Alternatively, perhaps have a function to split the name, perhaps with an engine name as an argument if different engines do different things.

Reply:
I will add a function that can parse the base task name and enforce the format of it.